transports.go

```diff
@@ -0,0 +1,30 @@
+package image
+
+import (
+	"fmt"
+
+	"github.com/containers/image/directory"
+	"github.com/containers/image/docker"
+	"github.com/containers/image/oci"
+	"github.com/containers/image/openshift"
+	"github.com/containers/image/types"
+)
+
+// KnownTransports is a registry of known ImageTransport instances.
+var KnownTransports map[string]types.ImageTransport
+
+func init() {
+	KnownTransports = make(map[string]types.ImageTransport)
+	for _, t := range []types.ImageTransport{
+		directory.Transport,
+		docker.Transport,
+		oci.Transport,
+		openshift.Transport,
+	} {
```

Comment:
This would nicely fit into `image`, but it needs to be a separate package because these transport-dependent packages depend on `image`, which would cause a circular dependency.

Adding a `containers/image/transports` for this small file feels a bit silly, and having the top-level `containers/image` package point at the contained transports seems to make sense — OTOH using both `containers/image` and `containers/image/image` in a single source file then ends up pretty very ugly.